src/schedule/zephyr_dp_schedule.c

```diff
@@ -131,7 +131,7 @@ static int scheduler_dp_task_free(void *data, struct task *task)
 	scheduler_dp_unlock(lock_key);
 
 	/* free task stack */
-	rfree(pdata->p_stack);
+	rfree(z_soc_uncached_ptr(pdata->p_stack));
```

Comment:
not needed, heap implementation works on both cached and uncached aliases. Changing to uncache here is just a waste of CPU cycles

Reply:
the same pointer should be used for freeing, that was allocated. If it's a cached pointer, so far our solution is to type-case it to uncached, e.g. like in https://github.com/thesofproject/sof/blob/1319163ea2c6d8bea6fd7189dea6275ac5e34bf5/src/audio/module_adapter/module_adapter.c#L403